hw1/task1/get_stats.py

```diff
@@ -0,0 +1,133 @@
+import pandas as pd
+import numpy as np
+
+
+def convert_type(type_to_convert, return_if_exception=None):
+    def return_function(x):
+        try:
+            x = type_to_convert(x)
+        except Exception:
+            return return_if_exception
+        return x
+    return return_function
+
+
+def get_time(datetime):
+    day_index = 0
+    datetime_str = str(datetime).split()
+    if datetime_str[day_index] == '0':
+        return datetime_str[-1]
+    else:
+        return str(datetime)
+
+
+def valid_duration(duration):
+    if duration < 0:
+        return None
+    else:
+        return duration
+
+
+def read_value_data(path):
+    data_without_header = pd.read_csv(path, skiprows=1, header=None)
+    header = pd.read_csv(path, nrows=0)
+
+    data_len = data_without_header.shape[1]
+    header_len = header.shape[1]
+
+    if data_len > header_len:
+        len_difference = data_len - header_len
+        columns_to_drop = list(range(data_len)[-len_difference:])
+
+        data = data_without_header.drop(columns_to_drop, axis=1)
+    elif data_len < header_len:
+        raise ValueError(f"Header length is {header_len}, but data length is {data_len}.")
+
+    data.columns = header.columns
+    data.columns = np.char.lower(np.array(data.columns, dtype=str))
+
+    required_columns = ['lpep_pickup_datetime',
+                        'lpep_dropoff_datetime',
+                        'passenger_count',
+                        'trip_distance',
+                        'total_amount',
+                        'trip_duration']
+
+    data['lpep_pickup_datetime'] = data['lpep_pickup_datetime'].apply(convert_type(np.datetime64))
+    data['lpep_dropoff_datetime'] = data['lpep_dropoff_datetime'].apply(convert_type(np.datetime64))
+    data['trip_duration'] = data['lpep_dropoff_datetime'] - data['lpep_pickup_datetime']
```

Comment:
when the file has only bad rows/rows mismatch with headers, script fails here with following error:
`TypeError: unsupported operand type(s) for -: 'DatetimeArray' and 'NoneType'`